grakn-kb/src/main/java/ai/grakn/kb/internal/concept/ElementFactory.java

```diff
@@ -204,20 +201,12 @@ public RoleImpl buildRole(VertexElement vertex, Role type){
      * @param edge A {@link Edge} of an unknown type
      * @return A concept built to the correct type
      */
-    public <X extends Concept> Optional<X> buildConcept(Edge edge){
+    public <X extends Concept> X buildConcept(Edge edge){
         return buildConcept(buildEdgeElement(edge));
     }
 
-    public <X extends Concept> Optional<X> buildConcept(EdgeElement edgeElement){
-        Schema.EdgeLabel label;
-
-        try {
-            label = Schema.EdgeLabel.valueOf(edgeElement.label().toUpperCase(Locale.getDefault()));
-        } catch (IllegalStateException e){
-            LOG.warn("Invalid edge [" + edgeElement + "] due to " + e.getMessage(), e);
-            return Optional.empty();
-        }
-
+    public <X extends Concept> X buildConcept(EdgeElement edgeElement){
+        Schema.EdgeLabel label = Schema.EdgeLabel.valueOf(edgeElement.label().toUpperCase(Locale.getDefault()));
```

Comment:
Is removing the `try...catch` deliberate?

Reply:
Yep now we just crash like badasses. BOOM!!!

Reply:
On a serious note it's just so we can fail when encountering weird edges rather than hide them.